modules/Kruskal_algorithm/src/Kruskal_algorithm.cpp

```diff
@@ -0,0 +1,103 @@
+// Copyright 2021 Volkova Anastasia
+
+#include <utility>
+#include <algorithm>
+#include <vector>
+#include <random>
+#include <ctime>
+
+#include "include/Kruskal_algorithm.h"
+
+Graph::Graph() : n(0) {
+    G.clear();
+    MST.clear();
+    parent.resize(0);
+}
+
+Graph::Graph(const int size) {
+    n = size;
+    parent.resize(n, 0);
+    G.clear();
+    MST.clear();
+    for (int i = 0; i < n; ++i) {
+        make_set(i);
+    }
+}
+
+Graph::Graph(const Graph& tmp) : n(tmp.get_n()),
+                                 G(tmp.get_G()),
+                                 MST(tmp.get_MST()),
+                                 parent(tmp.get_parent()) {}
+
+Graph& Graph::operator=(const Graph& tmp) {
+    n = tmp.get_n();
+    G = tmp.get_G();
+    MST = tmp.get_MST();
+    parent = tmp.get_parent();
+    return *this;
+}
```

Comment:
Проверка на самоприсваивание
```suggestion
Graph& Graph::operator=(const Graph& tmp) {
    if (this == &tmp) return *this;
    n = tmp.get_n();
    G = tmp.get_G();
    MST = tmp.get_MST();
    parent = tmp.get_parent();
    return *this;
}
```